integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala

```diff
@@ -71,52 +65,64 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
         Row("ALL", "ALL", 0L, 0L, 0L),
         Row(currentDatabase, "ALL", 0L, 0L, 0L))
     } else {
-      val carbonTables = CarbonEnv.getInstance(sparkSession).carbonMetaStore
-        .listAllTables(sparkSession).filter {
-        carbonTable =>
-          carbonTable.getDatabaseName.equalsIgnoreCase(currentDatabase) &&
-          isValidTable(carbonTable, sparkSession) &&
-          !carbonTable.isChildDataMap
+      val carbonTables = sparkSession.sessionState.catalog.listTables(currentDatabase).collect {
+        case tableIdent if CarbonEnv.getInstance(sparkSession).carbonMetaStore
+          .tableExists(tableIdent)(sparkSession) =>
+          CarbonEnv.getCarbonTable(tableIdent)(sparkSession)
```

Comment:
Its a two time lookup, instead directly call `CarbonEnv.getCarbonTable(tableIdent)(sparkSession)` and catch the excetion and ignore for non carbon tables

Reply:
Done,